RecoTauTag/RecoTau/interface/DeepTauBase.h

```diff
@@ -54,27 +82,25 @@ class DeepTauBase : public edm::stream::EDProducer<> {
     using OutputCollection = std::map<std::string, Output>;
 
 
-    DeepTauBase(const edm::ParameterSet& cfg, const OutputCollection& outputs);
-    virtual ~DeepTauBase();
+    DeepTauBase(const edm::ParameterSet& cfg, const OutputCollection& outputs, const DeepTauCache* cache);
+    virtual ~DeepTauBase() {}
 
     virtual void produce(edm::Event& event, const edm::EventSetup& es) override;
 
+    static std::unique_ptr<DeepTauCache> initializeGlobalCache(const edm::ParameterSet& cfg);
+    static void globalEndJob(const DeepTauCache* cache){ }
 private:
-    virtual tensorflow::Tensor GetPredictions(edm::Event& event, const edm::EventSetup& es,
+    virtual tensorflow::Tensor getPredictions(edm::Event& event, const edm::EventSetup& es,
                                               edm::Handle<TauCollection> taus) = 0;
-    virtual void CreateOutputs(edm::Event& event, const tensorflow::Tensor& pred, edm::Handle<TauCollection> taus);
+    virtual void createOutputs(edm::Event& event, const tensorflow::Tensor& pred, edm::Handle<TauCollection> taus);
 
 protected:
```

Comment:
Sorry for not commenting it earlier, but in CMS it is preferred (however not strictly required) to call data members with training `_` (my preference) or `m_`, e.g. `graphName` -> `graphName_` or `m_graphName`. 
It is also nice to use a consistent convention for multi-word names, e.g. with upper case for 2nd and following words, e.g `taus_token` -> `tausToken_`... 
It is also recommended to not use `_` as a first character (it happens at least one for a method argument in the implementation of DeepTauBase C-tor).